File: backend/flight_booking_project/src/main/java/com/javaweb/flight_booking_project/dto/request/ProfileRequest.java
package com.javaweb.flight_booking_project.dto.request;

import jakarta.validation.constraints.NotEmpty;
import lombok.Data;

import java.time.LocalDate;

@Data
public class ProfileRequest {
    @NotEmpty(message = "Name cannot empty")
    private String fullName;

    @NotEmpty(message = "Email cannot empty")
    private String email;

    @NotEmpty(message = "Phone cannot empty")
    private String phoneNumber;

    @NotEmpty(message = "Address cannot empty")
    private String address;

    private String dateOfBirth;
}

File: mobile/lib/data/model/profile_request.dart
class ProfileRequest{
  String fullName;
  String email;
  String phoneNumber;
  String address;
  String dateOfBirth;

  ProfileRequest({
    required this.fullName,
    required this.email,
    required this.phoneNumber,
    required this.address,
    required this.dateOfBirth,
  });

  factory ProfileRequest.fromJson(Map<String, dynamic> json) {
    return ProfileRequest(
      fullName: json["fullName"],
      email: json["email"],
      phoneNumber: json["phoneNumber"],
      address: json["address"],
      dateOfBirth: json["dateOfBirth"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "fullName": fullName,
      "email": email,
      "phoneNumber": phoneNumber,
      "address": address,
      "dateOfBirth": dateOfBirth,
    };
  }
}